tensorflow/python/ops/image_ops.py

```diff
@@ -1388,6 +1391,66 @@ def _crop_and_resize_shape(op):
 ops.RegisterShape('NonMaxSuppression')(common_shapes.call_cpp_shape_fn)
 
 
+def decode_image(contents, channels=None, ratio=None, fancy_upscaling=None,
+                 try_recover_truncated=None, acceptable_fraction=None,
+                 name=None):
+  """Convenience function for `decode_gif`, `decode_jpeg`, and `decode_png`.
+  Detects whether an image is a GIF, JPEG, or PNG, and performs the appropriate 
+  operation to convert the input bytes `string` into a `Tensor` of type `uint8`.
+
+  Note: `decode_gif` returns a 4-D array `[num_frames, height, width, 3]`, as 
+  opposed to `decode_jpeg` and `decode_png`, which return 3-D arrays 
+  `[height, width, num_channels]`. Make sure to take this into account when 
+  constructing your graph if you are intermixing GIF files with JPEG and/or PNG 
+  files.
+
+  Args:
+    contents: 0-D `string`. The encoded image bytes.
+    channels: Number of color channels for the decoded image.
+    ratio: Downscaling ratio (only used when decoding JPEG images)
+    fancy_upscaling: If true use a slower but nicer upscaling of the chroma 
+      planes (yuv420/422 JPEG images only).
+    try_recover_truncated: If true, try to recover an image from truncated input
+      (only used when decoding JPEG images).
+    acceptable_fraction: The minimum required fraction of lines before a 
+      truncated input is accepted (only used when decoding JPEG images).
+  
+  Returns:
+    `Tensor` with type `uint8`. Shape `[height, width, num_channels]` for JPEG 
+      and PNG images. Shape `[num_frames, height, width, 3]` for GIF images.
+  """
+  with ops.name_scope(name, 'decode_image') as scope:
+    def _gif():
+      return gen_image_ops.decode_gif(contents)
```

Comment:
I had to do the channels check with another assert message due to how `cond` runs it's function parameters. It seems like the Op-returning functions (`_gif`, `_png`,` _jpeg`) all run at Op creation time, which means that `_jpeg` gets unhappy when `channels` is 4 (granted, the user should probably just be using `decode_png` at that point). How does `if channels is not None or channels not in (0, 1, 3)`, throw an error at the beginning sound?